### PR TITLE
aws - ecs - adding network-location filter for ecs-service and ecs-task

### PR DESCRIPTION
**Pull request by nation-lucas**
This PR will resolve issue https://github.com/cloud-custodian/cloud-custodian/issues/8824 by adding network-location and security-group filters to aws.ecs-task and aws.ecs-service resources.

### CODE REVIEW

File: c7n/resources/ecs.py [hunk 1]
```diff
@@ -562,6 +584,41 @@ class TaskSubnetFilter(net_filters.SubnetFilter):
...
+        eni_ids = set()
+
+        cexp = jmespath_compile(self.eni_expression)
+        for r in resources:
+            ids = cexp.search(r)
+            if ids:
+                eni_ids.update(ids)
+
+        if eni_ids:
+            client = local_session(self.manager.session_factory).client('ec2')
```
Comment (kapilt) [inline c7n/resources/ecs.py:605]: we might need to cache this, get_related_ids gets called multiple times (first time is with full set of resources afaicr).

**Review by kapilt — COMMENTED** [review]
looks pretty good, but caching the eni api call would be good.

File: c7n/resources/ecs.py [hunk 2]
```diff
@@ -562,6 +584,58 @@ class TaskSubnetFilter(net_filters.SubnetFilter):
...
+            response = client.describe_network_interfaces(
+                NetworkInterfaceIds=list(eni_ids)
+            )
+            if response["NetworkInterfaces"]:
```
Comment (PratMis) [inline c7n/resources/ecs.py:612]: In scenarios where the `describe_network_interfaces` didn't go through, the `response["NetworkInterfaces"]` expression would error out.

**Comment by kapilt** [comment]
I think there might be some stale data files here, post the change to use caching.

**Comment by nation-lucas** [comment]
@kapilt Yep! I reran the tests with caching enabled.

**Review by PratMis — APPROVED** [review]
Thanks for all the updates. Lgtm

**Pull request merged**